doc/ecosystem.md

```diff
@@ -0,0 +1,131 @@
+provide additional functionality for users.
+
+*This page was adapted from [GeoPandas's Ecosystem](https://geopandas.org/en/latest/community/ecosystem.html) page.*
+
+## PyGMT dependencies
+
+### Required dependencies
+
+#### NumPy
+
+[NumPy][] is the fundamental package for scientific computing in Python. It is a Python
+library that provides a multidimensional array object, various derived objects (such as
+masked arrays and matrices), and an assortment of routines for fast operations on arrays,
+including mathematical, logical, shape manipulation, sorting, selecting, I/O, discrete
+Fourier transforms, basic linear algebra, basic statistical operations, random simulation
+and much more.
+
+#### pandas
+
+[pandas][] is a Python package providing fast, flexible, and expressive data structures
+designed to make working with "relational" or "labeled" data both easy and intuitive.
+It aims to be the fundamental high-level building block for doing practical, real-world
+data analysis in Python. Additionally, it has the broader goal of becoming the most
+powerful and flexible open source data analysis/manipulation tool available in any
+language. It is already well on its way toward this goal.
+
+#### Xarray
+
+[Xarray][] is an open source project and Python package that introduces labels in the
+form of dimensions, coordinates, and attributes on top of raw NumPy-like arrays, which
+allows for more intuitive, more concise, and less error-prone user experience.
+
+### Optional dependencies
+
+#### IPython
+
+[IPython][] provides a rich toolkit to help you make the most of using Python
+interactively. Its main components are a powerful interactive Python shell and a Jupyter
+kernel to work with Python code in Jupyter notebooks and other interactive frontends.
+
+PyGMT relies on IPython to provide a rich interactive experience in Jupyter notebooks.
+
+#### GeoPandas
+
+[GeoPandas][] is an open source project to make working with geospatial data in Python
+easier. GeoPandas extends the datatypes used by [pandas][] to allow spatial operations
+on geometric types. Geometric operations are performed by [Shapely][]. GeoPandas further
+depends on [pyogrio][] for file access and [Matplotlib][] for plotting.
+
+PyGMT doesn't directly rely on GeoPandas, but provides support of GeoPandas's data
+structure, {class}`geopandas.GeoDataFrame` and {class}`geopandas.GeoSeries`, which can
+be directly used in data processing and plotting functions/methods of PyGMT.
+
+#### contextily
+
+[contextily][] is a small Python package to retrieve tile maps from the internet. It can
+add those tiles as basemap to matplotlib figures or write tile maps to disk into
+geospatial raster files. Bounding boxes can be passed in both WGS84 (EPSG:4326) and
+Spheric Mercator (EPSG:3857).
+
+In PyGMT, {func}`pygmt.datasets.load_tile_map` and {class}`pygmt.Figure.tilemap` rely
+on it.
+
+#### rioxarray
+
+[rioxarray][] is a geospatial [Xarray][] extension powered by rasterio. Built on top of
```

Comment:
I can't remember where I found the rioxarray description. Maybe it's written by ChatGPT.